### PR TITLE
Update docs for VSO v0.5.0

### PR DESCRIPTION
**Pull request by benashz**
Doc updates for VSO v0.5.0
  
- Document the new secret transformation feature
- Update API and Helm reference docs

### CODE REVIEW

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+secret data output. The template data input holds the secret data, secret metadata, resource labels
+and annotations.
+
+VSO provides access to a large library of template functions, some of which are documented
+[below](#template-functions).
+
+### Secret data input
+
+Secret data is accessed through the `.Secrets` input member. It contains a map of secret key to
+its secret value. The data contains confidential information.
```
Comment (tvoran) [inline website/content/docs/platform/k8s/vso/secret-transformation.mdx:28]: Just thinking out loud about the last two sentences:

```suggestion
Secret data is accessed through the `.Secrets` input member. It contains a map of secret 
key-value pairs, which are assumed to be sensitive information fetched from a [secret source](/vault/docs/platform/k8s/vso/sources).
```

**Review by tvoran — APPROVED** [review]
Left some thoughts, but 👍

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+```yaml
+---
+apiVersion: secrets.hashicorp.com/v1beta1
+kind: SecretTransformation
+metadata:
+  name: vso-templates
+  namespace: example-vds
+spec:
+  excludes:
+    - password|username
```
[the remark this answers is not in the document]
Comment (benashz) [reply website/content/docs/platform/k8s/vso/secret-transformation.mdx:127]: > (Just curious) Is this equivalent to listing them individually?
> 
> ```yaml
> - password
> - username
> ```

Yes.

> What would an `&` operator do instead?

Not sure I follow. Would you mind providing an example use case?

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx (hunk 1 above)
Comment (benashz) [reply website/content/docs/platform/k8s/vso/secret-transformation.mdx]: Fixed in fc2d665d6fcae44c531c415e182b9a5b5318701b

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx [hunk 3]
```diff
@@ -0,0 +1,234 @@
+---
+layout: docs
+page_title: Vault Secrets Operator Secret Transformation
+description: >-
+  Secret data transformation.
```
[the remark this answers is not in the document]
Comment (benashz) [reply website/content/docs/platform/k8s/vso/secret-transformation.mdx]: Updated in fc2d665d6fcae44c531c415e182b9a5b5318701b

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    create: true
+    name: app-secret
+    transformation:
+      transformationRefs:
+        - name: vso-templates
+  path: creds/dev-postgres
+```
+
+The resulting Kubernetes secret includes a single key named `url`, with a valid Postgres connection
+URL as is value
```
[the remark this answers is not in the document]
Comment (benashz) [reply website/content/docs/platform/k8s/vso/secret-transformation.mdx]: Fixed in fc2d665d6fcae44c531c415e182b9a5b5318701b

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+      templates:
+        url:
+          text: |
+            {{- $host := get .Annotations "myapp.config/postgres-host" -}}
+            {{- printf "postgresql://%s:%s@%s/postgres?sslmode=disable" (get .Secrets "username") (get .Secrets "password") $host -}}
+  path: creds/dev-postgres
+```
+
+The resulting Kubernetes secret includes a single key named `url`, with a valid Postgres connection
+URL as is value.
```
[the remark this answers is not in the document]
Comment (benashz) [reply website/content/docs/platform/k8s/vso/secret-transformation.mdx]: Fixed in fc2d665d6fcae44c531c415e182b9a5b5318701b

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+For example, to include a value from the resource's labels in your application's secret, you
+might specify a template like:
+```
+{{- printf "appType=%s" (get .Labels "appType") -}}
+```
+
+## Filters
+
+Filters are used to control which source secret data fields are included in the destination secret's
+data. They specified as a set of exclude/include regular expressions, and are applied in that order.
```
[the remark this answers is not in the document]
Comment (benashz) [reply website/content/docs/platform/k8s/vso/secret-transformation.mdx]: Fixed in fc2d665d6fcae44c531c415e182b9a5b5318701b

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+### Secret data input
+
+Secret data is accessed through the `.Secrets` input member. It contains a map of secret key to
+its secret value. The data contains confidential information.
+
+For example, to include a password in your application's secret, you might specify
+a template like:
+```
+{{- printf "password=%s" (get .Secrets "password") -}}
+```
```
[the remark this answers is not in the document]
Comment (benashz) [reply website/content/docs/platform/k8s/vso/secret-transformation.mdx]: Updated in fc2d665d6fcae44c531c415e182b9a5b5318701b

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+might specify a template like:
+```
+{{- printf "appType=%s" (get .Labels "appType") -}}
+```
+
+## Filters
+
+Filters are used to control which source secret data fields are included in the destination secret's
+data. They specified as a set of exclude/include regular expressions, and are applied in that order.
+Filters are applied to the raw source data only.
```
[the remark this answers is not in the document]
Comment (benashz) [reply website/content/docs/platform/k8s/vso/secret-transformation.mdx]: Extended in fc2d665d6fcae44c531c415e182b9a5b5318701b

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+  Secret data transformation.
+---
+
+# Secret data transformation
+
+Utilizing advanced templating and data filters, the Vault Secrets Operator for Kubernetes (VSO) can
+transform source secret data, secret metadata, resource labels and annotations into a format that is
+compatible with your application. All secret data sources are supported. Secret transformations can
+be specified directly within a secret custom resource (CR), or by references to one or more
+`SecretTransformation` custom resource instances, or both.
```
[the remark this answers is not in the document]
Comment (benashz) [reply website/content/docs/platform/k8s/vso/secret-transformation.mdx]: Updated in fc2d665d6fcae44c531c415e182b9a5b5318701b

File: website/content/docs/platform/k8s/vso/secret-transformation.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tvoran) [reply website/content/docs/platform/k8s/vso/secret-transformation.mdx:138]: I don't have a use case, just wondering out loud if there's some reason for `[password|username]` vs `[password, username]`. If the intent here is to show that this accepts regex expressions, then I'd suggest something that isn't equivalent to listing them individually.

**Pull request merged**